packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -65,17 +66,20 @@ export function findOffsetInDOM(
 }
 
 export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
-  const nodeOffset = parseInt((node as HTMLElement).dataset.childBegin || '0', 10);
+  const nodeElement = node as HTMLElement;
+  const nodeOffset = parseInt(nodeElement.dataset.childBegin || '0', 10);
```

Comment:
Why this change is introduced? I think it can be as is previously.

Reply:
because i'm using this node element in many places below, instead of doing this `node as HTMLElement` every time, i think it's best to do it this way

Reply:
How about doing it here on line 68
```
export function getTextNodeAndOffset(node: HTMLElement, 
```

Reply:
because you don't always need it as an HTMLElement like when you pass it to the iterator.

Reply:
I don't really see the downside of having this variable.

Reply:
Oh, haven't seen that `node` is used in the iterator.  I am fine with the new variable then.